bridge/src/main/scala/xsbt/CompilerInterface.scala

```diff
@@ -50,12 +50,12 @@ class CachedCompilerImpl(args: Array[String], output: Output, resident: Boolean)
     (outputArgs ++ args.toList ++ sources.map(_.getAbsolutePath).sortWith(_ < _)).toArray[String]
 
   def run(sources: Array[File], changes: DependencyChanges, callback: AnalysisCallback, log: Logger, delegate: Reporter, progress: CompileProgress): Unit = synchronized {
-    run(sources.toList, changes, callback, log, progress)
+    run(sources.toList, changes, callback, log, delegate, progress)
   }
-  private[this] def run(sources: List[File], changes: DependencyChanges, callback: AnalysisCallback, log: Logger, compileProgress: CompileProgress): Unit = {
+  private[this] def run(sources: List[File], changes: DependencyChanges, callback: AnalysisCallback, log: Logger, delegate: Reporter, compileProgress: CompileProgress): Unit = {
     debug(log, args.mkString("Calling Dotty compiler with arguments  (CompilerInterface):\n\t", "\n\t", ""))
-    val ctx = (new ContextBase).initialCtx.fresh
-      .setSbtCallback(callback)
+    val freshContext = (new ContextBase).initialCtx.fresh
+    val ctx = freshContext.setSbtCallback(callback).setReporter(DelegatingReporter(delegate))
```

Comment:
OK, could you come up with a minimal example and open an issue? and leave a comment to indicate that you're working around a crash with the issue number